vowpalwabbit/search.cc

```diff
@@ -298,6 +300,59 @@ struct search_private
                   // the "real" decision step but this really only matters for caching purposes
   v_array<v_array<action_cache>*>
       memo_foreach_action;  // when foreach_action is on, we need to cache TRAIN trajectory actions for LEARN
+
+  ~search_private(){
+    if(all)
```

Comment:
nit: Can we make sure we're destroying stuff in the correct order?

Reply:
Once the v_array change comes in most of this destructor goes away so I'll leave this work to then